### PR TITLE
Implemented :meth:`.MovingCamera.auto_zoom` for automatically zooming onto specified mobjects

### PR DESCRIPTION
**Pull request by Gustav-Rixon**
## Overview: What does this pull request change?

Added auto_zoom to moving_camera.

Added a function to moving_camera called auto_zoom.

auto_zoom takes an array of Mobjects and calculates a new camera frame.

Fixed is_in_frame.

is_in_frame hade incorrect values it took the entire frame width/height from the center. It should be half of that.

## Motivation and Explanation: Why and how do your changes improve the library?

Solution to feature request #1056. Possibly a quality of life improvement to.

## Reviewer Checklist
- [ ] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

File: manim/camera/moving_camera.py [hunk 1]
```diff
@@ -9,9 +9,14 @@
...
 
+import math
+import operator
+
+from numpy import string_
+
 from .. import config
 from ..camera.camera import Camera
-from ..constants import ORIGIN
+from ..constants import *
```
Comment (behackl) [inline manim/camera/moving_camera.py:19]: I don't like *-imports in library files.
```suggestion
from ..constants import LEFT, RIGHT, UP, DOWN, ORIGIN
```

File: manim/camera/moving_camera.py [hunk 2]
```diff
@@ -173,3 +178,69 @@ def get_mobjects_indicating_movement(self):
         list
         """
         return [self.frame]
+
+    def auto_zoom(self, mobjects, margin=0, only_mobjects_in_frame=False):
+        """Zooms on to a given array of mobjects (or a singular mobject) and automatically resizes to frame all the mobjects
+
```
[the remark this answers is not in the document]
Comment (behackl) [reply manim/camera/moving_camera.py:186]: This has not been addressed.

File: tests/test_auto_zoom.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+def test_zoom():
+
+    s1 = Square()
+    s1.set_x(-10)
+    s2 = Square()
+    s2.set_x(10)
+
+    scene = MovingCameraScene()
+    scene.add(s1, s2)
+    scene.play(scene.camera.auto_zoom([s1, s2]))
```
Comment (behackl) [inline tests/test_auto_zoom.py:13]: We should make sure that nothing is actually rendered.

```suggestion
    with tempconfig({'dry_run': True}):
        scene = MovingCameraScene()
        scene.add(s1, s2)
        scene.play(scene.camera.auto_zoom([s1, s2]))
```

**Review by behackl — COMMENTED** [review]
Minor issues remain open for this PR; do you have time to fix them @Gustav-Rixon? I would like to have this PR included in the upcoming release.

File: manim/camera/moving_camera.py [hunk 4]
```diff
@@ -236,11 +239,11 @@ def auto_zoom(self, mobjects, margin=0, only_mobjects_in_frame=False):
...
+        new_Height = abs(scene_critical_y_up - scene_critical_y_down)
 
         # zoom to fit all mobjects along the side that has the largest size
-        if newWidth / self.frame.width > newHeight / self.frame.height:
-            return self.frame.animate.set_x(x).set_y(y).set(width=newWidth + margin)
+        if new_Width / self.frame.width > new_Height / self.frame.height:
+            return self.frame.animate.set_x(x).set_y(y).set(width=new_Width + margin)
         else:
-            return self.frame.animate.set_x(x).set_y(y).set(height=newHeight + margin)
+            return self.frame.animate.set_x(x).set_y(y).set(height=new_Height + margin)
```
[the remark this answers is not in the document]
Comment (Gustav-Rixon) [reply manim/camera/moving_camera.py]: a sorry i just misunderstood its fixed now.

File: manim/camera/moving_camera.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Gustav-Rixon) [reply manim/camera/moving_camera.py:180]: I added "This method only works when 2D-objects in the XY-plane are considered." above the method.

[the remark this answers is not in the document]
Comment (behackl) [reply manim/camera/moving_camera.py:180]: Please remove the comment above the method; this should be a proper note in the documentation:

```suggestion

        .. NOTE::

            This method only works when 2D-objects in the XY-plane are considered, it
            will not work correctly when the camera has been rotated.

```

File: tests/test_auto_zoom.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+
+def test_zoom():
+
+    s1 = Square()
+    s1.set_x(-10)
+    s2 = Square()
+    s2.set_x(10)
+
+    with tempconfig({"dry_run": True}):
```
Comment (behackl) [inline tests/test_auto_zoom.py:11]:
```suggestion
    with tempconfig({"dry_run": True, "quality": "low"}):
```

**Review by behackl — APPROVED** [review]
This looks good to me now, thank you!

File: tests/test_auto_zoom.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+
+def test_zoom():
+
+    s1 = Square()
+    s1.set_x(-10)
+    s2 = Square()
+    s2.set_x(10)
+
+    with tempconfig({"dry_run": True, "quality": "low"}):
```
Comment (behackl) [inline tests/test_auto_zoom.py:11]:
```suggestion
    with tempconfig({"dry_run": True, "quality": "low_quality"}):
```

**Pull request merged**